src/spikeinterface/comparison/groundtruthstudy.py

```diff
@@ -1,327 +1,405 @@
 from pathlib import Path
 import shutil
+import os
+import json
+import pickle
 
 import numpy as np
 
-from spikeinterface.core import load_extractor
-from spikeinterface.extractors import NpzSortingExtractor
-from spikeinterface.sorters import sorter_dict, run_sorters
+from spikeinterface.core import load_extractor, extract_waveforms, load_waveforms
+from spikeinterface.core.core_tools import SIJsonEncoder
+
+from spikeinterface.sorters import run_sorter_jobs, read_sorter_folder
 
 from spikeinterface import WaveformExtractor
 from spikeinterface.qualitymetrics import compute_quality_metrics
 
-from .paircomparisons import compare_sorter_to_ground_truth
+from .paircomparisons import compare_sorter_to_ground_truth, GroundTruthComparison
 
-from .studytools import (
-    setup_comparison_study,
-    get_rec_names,
-    get_recordings,
-    iter_working_folder,
-    iter_computed_names,
-    iter_computed_sorting,
-    collect_run_times,
-)
 
+# TODO later : save comparison in folders when comparison object will be able to serialize
 
-class GroundTruthStudy:
-    def __init__(self, study_folder=None):
-        import pandas as pd
 
-        self.study_folder = Path(study_folder)
-        self._is_scanned = False
-        self.computed_names = None
-        self.rec_names = None
-        self.sorter_names = None
+# This is to separate names when the key are tuples when saving folders
+_key_separator = " ## "
+# This would be more funny
+# _key_separator = " (°_°) "
+
+
+class GroundTruthStudy:
+    """
+    This class is an helper function to run any comparison on several "cases" for several ground truth dataset.
+
+    "cases" can be:
+      * several sorter for comparisons
+      * same sorter with differents parameters
+      * parameters of comparisons
+      * any combination of theses
+    
+    For enough flexibility cases key can be a tuple so that we can varify complexity along several
+    "levels" or "axis" (paremeters or sorter).
+
+    Generated dataframes will have index with several levels optionaly.
+    
+    Ground truth dataset need recording+sorting. This can be from mearec file or from the internal generator
+    :py:fun:`generate_ground_truth_recording()`
+    
+    This GroundTruthStudy have been refactor in version 0.100 to be more flexible than previous versions.
+    Folders structures are not backward compatible at all.
+    """
```

Comment:
```suggestion
    For increased flexibility, cases keys can be a tuple so that we can vary complexity along several
    "levels" or "axis" (paremeters or sorters).
    In this case, the result dataframes will have `MultiIndex` to handle the different levels.
    
    A ground-truth dataset is made of a `Recording` and a `Sorting` object. For example, it can be a simulated dataset with MEArec or internally generated (see 
    :py:fun:`~spikeinterface.core.generate.generate_ground_truth_recording()`).
    
    This GroundTruthStudy have been refactor in version 0.100 to be more flexible than previous versions.
    Note that the underlying folder structure is not backward compatible!
    """
```